### DISCUSSION TITLE
deterministic=True requires SQLite 3.8.3 or higher

### DISCUSSION THREAD

**Discussion by andekx** [vercel/community#2256] [Help]
### Summary

Hi everyone! I am deploying a Django project (web aplication) in Vercel and I was testing the application on my local machine only, so now I am strugling with the deployment since I never did it before. I know that Vercel doesn't support database so I tried to link my Vercel account with PlanetScale account and I believe I did it right, but I am still getting the error message bellow:

![image](https://user-images.githubusercontent.com/92690726/235536018-985df373-12da-4e98-9657-bbfed5becc7d.png)

Is anycode missing to sync between the sites or something else? I very new to programming so that's probably something very dumb that I forgot to do.

Appreciate any help.

**Comment by amyegan** [comment]
I'm sorry you're having trouble with your project, @andekx. Can you walk us through the steps you took to connect your database and what you have already tried while debugging this problem? If you're also able to share the repo or a [minimal reproduction](https://vercel.com/support/articles/creating-a-minimal-reproducible-example), that is usually very helpful for collaborative debugging.

**Reply by andekx** [reply]
Hey Amy! Thanks for the reply!

First for the integration I did:

1- Opened my project in Vercel and wen to Settings;
2- Integrations;
3- Searched for PlanetScale and add the integration.

So the configuration is something like this:
![image](https://user-images.githubusercontent.com/92690726/236345378-11a1cf20-3e88-44cb-9081-80babfb0f169.png)

![image](https://user-images.githubusercontent.com/92690726/236345402-ac2eee0d-59b5-47aa-8a0b-dbdf5a27d789.png)

Except that I did nothing more, I am guessing I need to make the migrations of my database but I have no clue, as I said I totally new in programming, sorry.

**Reply by Siddhu2502** [reply]
@amyegan it will be really helpful if this problem's solution is stated i am facing the same issue and there seems to be no solution even on the internet or even stack overflow (there are some ways of manually overriding the values but then those can be done in local machine but not on deployment level) so kindly look into it and help fix this problem and also @andekx if you have found a fix please give me the solution

**Reply by amyegan** [reply]
I have not seen your exact project setup, and I don't want to make assumptions, but this seems like a Django error from what I can see. The PlanetScale integration adds the necessary environment variables, but you'll need to make sure your project is set up appropriately. 

I found a few resources which may help you figure out what is going wrong for your site:
- [Is SQLite supported in Vercel?](https://vercel.com/guides/is-sqlite-supported-in-vercel)
- [Installing something else in the build image](https://vercel.com/docs/concepts/deployments/build-image#installing-something-else-in-the-build-image)
- [Django - deterministic=True requires SQLite 3.8.3 or higher upon running python manage.py runserver](https://stackoverflow.com/questions/66380006/django-deterministic-true-requires-sqlite-3-8-3-or-higher-upon-running-python)
- https://github.com/simonw/sqlite-utils/issues/425
- https://github.com/simonw/sqlite-utils/issues/408
- [PlanetScale site starter templates](https://vercel.com/templates?database=planetscale)

**Reply by VishalYadavGit** [reply]
Did you find the solution as I am struggling with the same problem and not able to find a solution?

Edit: I found the reason, So basically in vercel you can't host a SQLite DB, You will have to host postgres in the vercel or somewhere else.

**Reply by michealstarx** [reply]
I also need help guys

**Comment by VishalYadavGit** [comment]
> ### Summary
> Hi everyone! I am deploying a Django project (web aplication) in Vercel and I was testing the application on my local machine only, so now I am strugling with the deployment since I never did it before. I know that Vercel doesn't support database so I tried to link my Vercel account with PlanetScale account and I believe I did it right, but I am still getting the error message bellow:
> 
> ![image](https://user-images.githubusercontent.com/92690726/235536018-985df373-12da-4e98-9657-bbfed5becc7d.png)
> 
> Is anycode missing to sync between the sites or something else? I very new to programming so that's probably something very dumb that I forgot to do.
> 
> Appreciate any help.
> 
> ### Additional information
> _No response_
> 
> ### Example
> _No response_

So basically you can't host sqlite in vercel you will have to use postgres which should be hosted in vercel db or any other platform.
